lib/brains/forms.py

```diff
@@ -238,6 +241,35 @@ def clean(self):
             raise forms.ValidationError(
                 'Either nonce or paymethod must be set', code='invalid')
 
+        # The only way I can get the mock to stick is to do a late import.
+        # If we can fix that we can move this up.
+        from payments_config import products
+        product = products.get(self.cleaned_data.get('product_id'))
+
+        if not product:
+            raise forms.ValidationError(
+                'Product does not exist: {}'
+                .format(self.cleaned_data.get('product_id')),
+                code='invalid')
+
+        if (product.amount
+                and product.amount != self.cleaned_data.get('amount')):
```

Comment:
So my logic will fail when `amount` is not specified in the form, but is specified on the product. Yeah I see, will fix.